typhos/tests/conftest.py

```diff
@@ -129,8 +129,15 @@ def pytest_runtest_call(item: pytest.Item):
 
     t0 = time.monotonic()
     while time.monotonic() - t0 < 1.0 and len(_dereference_list(ending_widgets)) > 0:
-        gc.collect()
+        if not list(item.iter_markers(name="no_gc")):
+            # Some tests segfault on gc.collect when QT_QPA_PLATFORM=offscreen (!?)
+            # Mark these with pytest.mark.no_gc to skip this call
+            # Note that some other tests definitely need this gc.collect() call!
+            # And some tests function just fine no matter whether we call this or not!
+            gc.collect()
         app.processEvents()
+        # Throttle while loop to avoid weird gc things
+        time.sleep(0.1)
```

Comment:
This worries me, given inevitable hardware differences on the CI runners.

Reply:
I think the size of the sleep timer here didn't matter so much, the important part was to give background threads some time to process in cases where `app.processEvents()` is short rather than repeatedly calling `gc.collect()` in a no-wait loop.